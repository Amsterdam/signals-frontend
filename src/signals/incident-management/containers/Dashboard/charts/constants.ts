// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam

export const timeFormatLocale = {
  dateTime: '%a %e %B %Y %X',
  date: '%d-%m-%Y',
  time: '%H:%M:%S',
  periods: ['AM', 'PM'],
  days: [
    'zondag',
    'maandag',
    'dinsdag',
    'woensdag',
    'donderdag',
    'vrijdag',
    'zaterdag',
  ],
  shortDays: ['Zo', 'Ma', 'Di', 'Wo', 'Do', 'Vr', 'Za'],
  months: [
    'januari',
    'februari',
    'maart',
    'april',
    'mei',
    'juni',
    'juli',
    'augustus',
    'september',
    'oktober',
    'november',
    'december',
  ],
  shortMonths: [
    'jan',
    'feb',
    'mrt',
    'apr',
    'mei',
    'jun',
    'jul',
    'aug',
    'sep',
    'okt',
    'nov',
    'dec',
  ],
}
