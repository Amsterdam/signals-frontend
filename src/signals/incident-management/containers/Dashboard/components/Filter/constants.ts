// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam
export const filterNames = [
  'department',
  'category',
  'priority',
  'punctuality',
  'district',
]
