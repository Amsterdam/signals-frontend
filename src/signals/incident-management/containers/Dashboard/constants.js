export const REQUEST_DASHBOARD = 'src/Dashboard/REQUEST_DASHBOARD';
export const REQUEST_DASHBOARD_SUCCESS = 'src/Dashboard/REQUEST_DASHBOARD_SUCCESS';
