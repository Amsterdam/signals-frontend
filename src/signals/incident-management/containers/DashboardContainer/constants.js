export const REQUEST_DASHBOARD = 'src/DashboardContainer/REQUEST_DASHBOARD';
export const REQUEST_DASHBOARD_SUCCESS = 'src/DashboardContainer/REQUEST_DASHBOARD_SUCCESS';
export const REQUEST_DASHBOARD_ERROR = 'src/DashboardContainer/REQUEST_DASHBOARD_ERROR';
