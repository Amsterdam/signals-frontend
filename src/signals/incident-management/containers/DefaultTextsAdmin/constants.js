export const FETCH_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS';
export const FETCH_DEFAULT_TEXTS_SUCCESS = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS_SUCCESS';
export const FETCH_DEFAULT_TEXTS_ERROR = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS_ERROR';

export const STORE_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS';
export const STORE_DEFAULT_TEXTS_SUCCESS = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS_SUCCESS';
export const STORE_DEFAULT_TEXTS_ERROR = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS_ERROR';

export const ORDER_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/ORDER_DEFAULT_TEXTS';
