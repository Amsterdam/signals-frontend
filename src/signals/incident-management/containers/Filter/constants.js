export const SAVE_FILTER = 'sia/Filter/SAVE_FILTER';
export const SAVE_FILTER_SUCCESS = 'sia/Filter/SAVE_FILTER_SUCCESS';
export const SAVE_FILTER_FAILED = 'sia/Filter/SAVE_FILTER_FAILED';

export const UPDATE_FILTER = 'sia/Filter/UPDATE_FILTER';
export const UPDATE_FILTER_SUCCESS = 'sia/Filter/UPDATE_FILTER_SUCCESS';
export const UPDATE_FILTER_FAILED = 'sia/Filter/UPDATE_FILTER_FAILED';

export const GET_FILTERS = 'sia/Filter/GET_FILTERS';
export const GET_FILTERS_SUCCESS = 'sia/Filter/GET_FILTERS_SUCCESS';
export const GET_FILTERS_FAILED = 'sia/Filter/GET_FILTERS_FAILED';

export const CLEAR_FILTER = 'sia/Filter/CLEAR_FILTER';
