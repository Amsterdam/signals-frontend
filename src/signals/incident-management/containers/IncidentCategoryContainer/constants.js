export const REQUEST_CATEGORY_UPDATE = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE';
export const REQUEST_CATEGORY_UPDATE_SUCCESS = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE_SUCCESS';
export const REQUEST_CATEGORY_UPDATE_ERROR = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE_ERROR';
