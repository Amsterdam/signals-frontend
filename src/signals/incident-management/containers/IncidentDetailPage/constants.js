export const REQUEST_INCIDENT = 'src/IncidentDetailPage/REQUEST_INCIDENT';
export const REQUEST_INCIDENT_SUCCESS = 'src/IncidentDetailPage/REQUEST_INCIDENT_SUCCESS';
export const REQUEST_INCIDENT_ERROR = 'src/IncidentDetailPage/REQUEST_INCIDENT_ERROR';

export const REQUEST_NOTES_LIST = 'src/IncidentDetailPage/REQUEST_NOTES_LIST';
export const REQUEST_NOTES_LIST_SUCCESS = 'src/IncidentDetailPage/REQUEST_NOTES_LIST_SUCCESS';
export const REQUEST_NOTES_LIST_ERROR = 'src/IncidentDetailPage/REQUEST_NOTES_LIST_ERROR';
