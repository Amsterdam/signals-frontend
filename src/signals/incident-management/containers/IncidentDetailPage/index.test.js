import React from 'react';
import { shallow } from 'enzyme';

import { IncidentDetailPage, mapDispatchToProps } from './index';
import { REQUEST_INCIDENT } from './constants';
import stadsdeelList from '../../definitions/stadsdeelList';


describe('<IncidentDetailPage />', () => {
  let props;

  beforeEach(() => {
    props = {
      incidentdetailpage: { incident: {}, stadsdeelList },
      onRequestIncident: jest.fn()
    };
  });

  it('should render correctly', () => {
    const renderedComponent = shallow(
      <IncidentDetailPage {...props} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    // For the `mapDispatchToProps`, call it directly but pass in
    // a mock function and check the arguments passed in are as expected
    mapDispatchToProps(dispatch).onRequestIncident({});
    expect(dispatch.mock.calls[0][0]).toEqual({ type: REQUEST_INCIDENT, payload: {} });
  });
});
