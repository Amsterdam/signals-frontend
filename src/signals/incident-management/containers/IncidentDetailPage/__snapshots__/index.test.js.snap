// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentDetailPage /> should render correctly 1`] = `
<div
  className="incident-detail-page row container"
>
  <div
    className="col-12"
  >
    <h3>
      Melding 
      100
    </h3>
  </div>
  <ul
    className="col-12 col-md-4 incident-detail-page__map"
  >
    <MapDetail
      label=""
    />
  </ul>
  <div
    className="col-12 col-md-8"
  >
    (
    <Link
      replace={false}
      to="undefined/incidents"
    >
      Terug naar overzicht
    </Link>
    )
    <button
      onClick={[Function]}
    >
      Print view
    </button>
    <IncidentDetail
      incident={Object {}}
      priorityList={
        Array [
          Object {
            "key": "",
            "value": "Alles",
          },
          Object {
            "key": "normal",
            "value": "Normaal",
          },
          Object {
            "key": "high",
            "value": "Hoog",
          },
        ]
      }
      stadsdeelList={
        Array [
          Object {
            "key": "",
            "value": "Alle stadsdelen",
          },
          Object {
            "key": "A",
            "value": "Centrum",
          },
          Object {
            "key": "B",
            "value": "Westpoort",
          },
          Object {
            "key": "E",
            "value": "West",
          },
          Object {
            "key": "M",
            "value": "Oost",
          },
          Object {
            "key": "N",
            "value": "Noord",
          },
          Object {
            "key": "T",
            "value": "Zuidoost",
          },
          Object {
            "key": "K",
            "value": "Zuid",
          },
          Object {
            "key": "F",
            "value": "Nieuw-West",
          },
        ]
      }
    />
  </div>
  <div
    className="col-12"
  >
    <Tabs
      onTabChanged={[Function]}
      selectedTab={0}
      tabs={
        Array [
          Object {
            "name": "Status",
            "value": <withReducer(withSaga(Connect(IncidentStatusContainer)))
              id="100"
            />,
          },
          Object {
            "name": "Urgentie",
            "value": <withReducer(withSaga(Connect(IncidentPriorityContainer)))
              id="100"
            />,
          },
          Object {
            "name": "Subcategorie",
            "value": <withReducer(withSaga(Connect(IncidentCategoryContainer)))
              id="100"
            />,
          },
          Object {
            "count": undefined,
            "name": "Notities",
            "value": <withReducer(withSaga(Connect(IncidentNotesContainer)))
              id="100"
            />,
          },
        ]
      }
    />
  </div>
  <div
    className="col-12"
  >
    <div
      className="incident-detail-page__tab-container"
    >
      <withReducer(withSaga(Connect(IncidentStatusContainer)))
        id="100"
      />
    </div>
  </div>
</div>
`;

exports[`<IncidentDetailPage /> should render the print view 1`] = `
<Connect(PrintLayout)
  id="100"
  incident={Object {}}
  onPrintView={[Function]}
  priorityList={
    Array [
      Object {
        "key": "",
        "value": "Alles",
      },
      Object {
        "key": "normal",
        "value": "Normaal",
      },
      Object {
        "key": "high",
        "value": "Hoog",
      },
    ]
  }
  stadsdeelList={
    Array [
      Object {
        "key": "",
        "value": "Alle stadsdelen",
      },
      Object {
        "key": "A",
        "value": "Centrum",
      },
      Object {
        "key": "B",
        "value": "Westpoort",
      },
      Object {
        "key": "E",
        "value": "West",
      },
      Object {
        "key": "M",
        "value": "Oost",
      },
      Object {
        "key": "N",
        "value": "Noord",
      },
      Object {
        "key": "T",
        "value": "Zuidoost",
      },
      Object {
        "key": "K",
        "value": "Zuid",
      },
      Object {
        "key": "F",
        "value": "Nieuw-West",
      },
    ]
  }
/>
`;
